src/renderer/components/cluster-manager/cluster-frame-handler.ts

```diff
@@ -88,7 +88,15 @@ export class ClusterFrameHandler {
           () => {
             this.dependencies.logger.info(`[LENS-VIEW]: remove dashboard, clusterId=${clusterId}`);
             this.views.delete(clusterId);
-            parentElem.removeChild(iframe);
+
+            // Must only remove iframe from DOM after it unloads old code. Else it crashes
+            iframe.addEventListener("load", () => parentElem.removeChild(iframe), {
```

Comment:
`"load"`? How does this work?

Reply:
That event even fires when the load fails.

Reply:
I guess I don't see where the `"load"` event gets called in this situation. By `iframe.setAttribute("src", "");`?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event

This means that the DOM and all script resources have been loaded.

An empty src means "no src" or empty since that is the default.